tests/assets/dashAgGridFunctions.js

```diff
@@ -171,4 +171,52 @@ function monthToComparableNumber(date) {
   const monthNumber = parseInt(date.split('/')[1]);
   const dayNumber = parseInt(date.split('/')[0]);
   return yearNumber * 10000 + monthNumber * 100 + dayNumber;
-}
\ No newline at end of file
+}
+
+const {useImperativeHandle, useState, useEffect, forwardRef} = React;
+
+dagfuncs.YearFilter = forwardRef((props, ref) => {
```

Comment:
```suggestion
// This example was adapted from https://www.ag-grid.com/react-data-grid/component-filter/
// The only differences are:
// - React.createElement instead of JSX
// - setProps, which all Dash components use to report user interactions,
//     instead of a plain js event handler
dagfuncs.YearFilter = forwardRef((props, ref) => {
```